src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processor/PersonService.kt

```diff
@@ -17,34 +16,38 @@ class PersonService(
   private val telemetryService: TelemetryService,
 ) {
 
-  fun processPerson(person: Person) {
-    val existingPersonEntity: PersonEntity? = getPersonEntityBySourceSystem(person)
-    handlePerson(existingPersonEntity, person)
+  fun processPerson(person: Person, callback: () -> PersonEntity?) {
+    val existingPersonEntity: PersonEntity? = callback()
+    handlePerson(person, existingPersonEntity)
```

Comment:
👏

Reply:
From this point onwards it is a big improvement. Do we actually need a callback here though? What do we gain over passing a parameter of type`PersonEntity?`

Reply:
I also thought of the same thing. Since we are not doing anything more in the callback, we could just pass the parameter. Happy either way.